### ISSUE TITLE
Threaded resource loader loses textures

### ISSUE THREAD

**Issue by BastiaanOlij** [godotengine/godot#77416]
### Godot version

4.0.dev (f581f21dd61a8fb581b80d07755cdf60c95d146d)

### System information

Windows 10 - Ryzen 7 - 4070ti

### Issue description

When using the threaded resource loader, textures on materials are lost.

This is the scene being loaded in the IDE:
![image](https://github.com/godotengine/godot/assets/1945449/232dd16b-81c4-49cf-a980-3efcfa97ff79)

This is the result in runtime:
![image](https://github.com/godotengine/godot/assets/1945449/235176d3-6058-4bed-8fbe-cad14a5e0d32)

### Steps to reproduce

Use code such as the following to load a scene:
```
func _load_scene(p_scene_path):
	var new_scene : PackedScene
	if ResourceLoader.has_cached(p_scene_path):
		# Load cached scene
		new_scene = ResourceLoader.load(p_scene_path)
	else:
		# Start the loading in a thread
		ResourceLoader.load_threaded_request(p_scene_path)

		# Loop waiting for the scene to load
		while true:
			var progress := []
			var res := ResourceLoader.load_threaded_get_status(p_scene_path, progress)
			if res != ResourceLoader.THREAD_LOAD_IN_PROGRESS:
				break;

			_progress_bar = progress[0]
			await get_tree().create_timer(0.1).timeout

		# Get the loaded scene
		new_scene = ResourceLoader.load_threaded_get(p_scene_path)
	
	$CanvasLayer.visible = false
	add_child(new_scene.instantiate())
```

Now create a scene to load that has a material with a texture.

### Minimal reproduction project

[TestResourceLoader.zip](https://github.com/godotengine/godot/files/11549692/TestResourceLoader.zip)

**Comment by BastiaanOlij** [comment]
From my own investigation so far my theory is that the textures loaded from a thread only get their RID reserved in the rendering server. The method calls that actually setup the texture are added to the command queue of the rendering server thread (/main thread). 

When the material is loaded it may be having issues accessing vital information about the texture as the queued commands haven't been run yet.

Could be barking up the wrong tree though, still trying to get my head around the threading changes :)

**Comment by Anixias** [comment]
May be a different issue, and it's 2D instead of 3D, but when I used the threaded resource loader, it loses materials. For instance, my creatures' shadows no longer had a multiply material, and tiles in the tilesets no longer have their custom materials as well.

**Comment by FeralBytes** [comment]
I can confirm, running the same project in 4.0.3 using threaded loading results in normal colors. But running it in 4.1.dev3 results in all the colors disappearing from all objects.

**Comment by sketchyfun** [comment]
> May be a different issue, and it's 2D instead of 3D, but when I used the threaded resource loader, it lost materials. For instance, my creatures' shadows no longer had a multiply material, and tiles in the tilesets no longer have their custom materials as well.

Yeah, I'm noticing this with a dev build I built yesterday. 

Without threaded loading (or just running directly from the editor):
![image](https://github.com/godotengine/godot/assets/10464013/a1683f19-b602-4aed-b107-ed7575794357)

With threaded loading:
![image](https://github.com/godotengine/godot/assets/10464013/76da5cc0-73be-4b09-be54-073a014eac79)

**Comment by BastiaanOlij** [comment]
@Anixias this would likely effect 2D too so not surprised

@FeralBytes yes this is on master from a few days ago and it is indeed a regression from the threaded resource loader. @RandomShaper was going to look into it as soon as he has some time.

@sketchyfun the editor also turns threaded loading off hence it working in the editor so that makes total sense.

**Comment by BastiaanOlij** [comment]
Confirmed fixed by #77692

**Comment by BluntBSE** [comment]
I am still getting this issue in Godot 4.3, actually.
